components/places/src/storage/history.rs

```diff
@@ -155,6 +155,18 @@ pub fn apply_observation_direct(
     Ok(visit_row_id)
 }
 
+pub fn update_frecencies(db: &PlacesDb) -> Result<()> {
```

Comment:
This also isn't ideal.. and doesn't **need** to happen during the migration, also looking for more efficient ways. I'm thinking a possibility is to even have it run async on its own, or instead, set an arbitrary non `-1` value 🤷 

I added it because with `-1` frecency the queryAutcomplete is acting weird and not even displaying values correctly (it's not searching the titles and only searching the URLs)

Reply:
Yeah, that logic seems precisely what I want.

One thing to consider though is that iOS only allows there to exist one writer connection, so writers will actually be blocked as the whole `moz_places_stale_frecencies` process runs. This is unfortunate because the bookmark syncing logic was chunked so that other writer connections could sneak in. 

That said, as long as it's in a different transaction, the reader connection can go ahead and read the history (although with stale frecencies which is OK since they'll be updated anyways)

I'll change this so it still runs as a part of the migration process, but it runs in a separate transaction properly